shotover-proxy/src/codec/redis.rs

```diff
@@ -7,9 +7,16 @@ use redis_protocol::resp2::prelude::decode_mut;
 use redis_protocol::resp2::prelude::encode_bytes;
 use tokio_util::codec::{Decoder, Encoder};
 
+#[derive(Debug, Clone)]
+pub enum RedisDirection {
```

Comment:
So my understanding is we use this for perf reasons? 

instead of first checking whether something is a Redis command and then checking it's validity.

Reply:
My thinking is that we could get rid of this enum and just check everything on the codec methods. Assuming the sink won't be sending us commands that is.

Reply:
We can't always check because the server responses are not valid commands

Reply:
Ah my bad, I was thinking in Cassandra land where we have op codes.